accounts-db/src/append_vec.rs

```diff
@@ -1034,10 +1034,18 @@ impl AppendVec {
                     offset = next.next_account_offset;
                 }
             }
-            AppendVecFileBacking::File(_file) => {
-                self.scan_accounts(|stored_meta| {
-                    callback(stored_meta.pubkey());
-                });
+            AppendVecFileBacking::File(file) => {
+                let buffer_size = std::cmp::min(SCAN_BUFFER_SIZE, self.len());
```

Comment:
SCAN_BUFFER_SIZE is around 30MB.  And the data, which we are really interested in, is 136B.  It seems like we are over fetching the buffer and don't skip much of the account's data. 

For example, let's say that account data.len is about 100K. Then, each time we read, we fetch up to 30 accounts in the buffer. And we will only skip 1 account after we used all 30 accounts in the buffer. then fetch another 30 Accounts... Only 1 out of 30 accounts are skipped.

How about making 'SCAN_BUFFER_SIZE' smaller, i.e. 1K?
For the example above, 1K buffer will nicely skip loading 99K data of the accounts.

Reply:
Maybe 1M is better. There is a tradeoff, considering it would be more efficient to read larger buffer at a time.

Reply:
wdyt?

Reply:
Probably we can write a benchmark: generate 20K account, each account data 1k-10K data and try different buffer size?

Reply:
Also, it seems that we can reduce the size of the buffer  for `scan_index` too.

Reply:
ok. sounds good to me.